packages/lodestar/src/chain/regen/interface.ts

```diff
@@ -9,21 +43,28 @@ export interface IStateRegenerator {
    * Return a valid pre-state for a beacon block
    * This will always return a state in the latest viable epoch
    */
-  getPreState: (block: allForks.BeaconBlock) => Promise<CachedBeaconState<allForks.BeaconState>>;
+  getPreState: (block: allForks.BeaconBlock, rCaller: IRegenCaller) => Promise<CachedBeaconState<allForks.BeaconState>>;
 
   /**
    * Return a valid checkpoint state
    * This will always return a state with `state.slot % SLOTS_PER_EPOCH === 0`
    */
-  getCheckpointState: (cp: phase0.Checkpoint) => Promise<CachedBeaconState<allForks.BeaconState>>;
+  getCheckpointState: (
+    cp: phase0.Checkpoint,
+    rCaller: IRegenCaller
```

Comment:
You don't need to know the function name in `IRegenCaller`, this function signature can just be

```ts
getCheckpointState: (cp: phase0.Checkpoint, caller: RegenCaller)
```